build.rs

```diff
@@ -9,7 +9,10 @@ use std::{
     str::FromStr,
 };
 
-const PY3_MIN_MINOR: u8 = 5;
+/// Minimum required Python version.
+const PY3_MIN_MINOR: u8 = 6;
+/// Maximum Python version that can be used as minimum required Python version with abi3.
+const ABI3_MAX_MINOR: u8 = 9;
```

Comment:
Binaries trying to link against a higher python version than the actual installed `libpython` will fail because symbols will mismatch. So this new feature can only work for extension modules. 

(And even worse, only extension modules on linux - because on windows they always link!)

This would mean that an extension module which uses the `abi3-pyXY` features would further break `cargo test` even after #1123 , and would also not be cross-platform.